pkg/utils/gardener/seed.go

```diff
@@ -152,7 +152,7 @@ func GetWildcardCertificate(ctx context.Context, c client.Client) (*corev1.Secre
 }
 
 // SeedIsGarden returns 'true' if the cluster is registered as a Garden cluster.
-func SeedIsGarden(ctx context.Context, seedClient client.Client) (bool, error) {
+func SeedIsGarden(ctx context.Context, seedClient client.Reader) (bool, error) {
```

Comment:
Why are we using `client.Reader`? 1 min should be enough for the cache to get synced.

Reply:
In the scope of this function, `client.Reader` is sufficient since only read requests are performed by it. If you ask why we use `client.APIReader` when invoking this function in the new code: This is because the controller-runtime manager might not be started at the time when we need to know whether the seed is garden, hence its caches are not synced, hence its `client.Client` cannot be used.

Reply:
> If you ask why we use client.APIReader when invoking this function in the new code:

I wanted to ask this.
> This is because the controller-runtime manager might not be started at the time when we need to know whether the seed is garden, hence its caches are not synced, hence its client.Client cannot be used.

Ok, make sense 👍